glean-core/src/lib.rs

```diff
@@ -127,8 +129,17 @@ impl Glean {
         let ping_maker = PingMaker::new();
         let doc_id = Uuid::new_v4().to_string();
         let url_path = self.make_path(ping_name, &doc_id);
-        let ping_content =
-            ::serde_json::to_string_pretty(&ping_maker.collect(self.storage(), ping_name)).unwrap();
+        let ping_content = match ping_maker.collect_string(self.storage(), ping_name) {
```

Comment:
Can we add unit test coverage to make sure that `make_path` behaves as expected?